File: app/src/main/java/com/example/peyo/gles20/Triangle.java
package com.example.peyo.gles20;

import android.opengl.GLES20;

public class Triangle extends GLES20 {

    static float coords[] = {
            0, 1, -1,  -1, -1, 1,  1, -1, 1  };

    static float colors[] = {
            0, 1, 0, 1,  0, 0, 1, 1,  1, 0, 0, 1 };

    private int vertexVBO[] = new int[1];
    private int colorVBO[] = new int[1];

    private int mPositionHandle;
    private int mColorHandle;

    public Triangle(int position, int color) {
        GLToolbox.loadFloatVBO(vertexVBO, coords);
        GLToolbox.loadFloatVBO(colorVBO, colors);

        mPositionHandle = position;
        mColorHandle = color;
    }

    public void draw() {
        glBindBuffer(GL_ARRAY_BUFFER, vertexVBO[0]);
        glVertexAttribPointer(mPositionHandle, 3, GL_FLOAT, false, 3*4, 0);
        glBindBuffer(GL_ARRAY_BUFFER, colorVBO[0]);
        glVertexAttribPointer(mColorHandle, 4, GL_FLOAT, false, 4*4, 0);
        glBindBuffer(GL_ARRAY_BUFFER, 0);

        glDrawArrays(GL_TRIANGLES, 0, 3);
    }
}

File: app/src/main/java/com/example/peyo/gles20/Triangle.kt
package com.example.peyo.gles20

import android.opengl.GLES20
import java.nio.FloatBuffer

val triangleCoords = floatArrayOf (
        0f, 2f, 0f,
        -1f, -1f, 0f,
        1f, -1f, 0f)

val triangleColors = floatArrayOf (
        0f, 1f, 0f, 1f,
        0f, 0f, 1f, 1f,
        1f, 0f, 0f, 1f )


class Triangle(val positionHandle: Int, val colorHandle: Int) : GLES20() {

    private val vertexCount: Int = triangleCoords.size / 3
    private val vertexBuffer: FloatBuffer = GLToolbox.loadBuffer(triangleCoords)
    private val colorBuffer: FloatBuffer = GLToolbox.loadBuffer(triangleColors)

    fun draw() {
        glVertexAttribPointer(positionHandle, 3, GL_FLOAT, false,
                3 * 4, vertexBuffer)
        glVertexAttribPointer(colorHandle, 4, GL_FLOAT, false,
                4 * 4, colorBuffer)
        glDrawArrays(GL_TRIANGLES, 0, vertexCount)
    }
}